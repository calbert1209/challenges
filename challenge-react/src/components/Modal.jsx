import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  position: absolute;
  background-color: hsla(0, 0%, 100%, 0.904);
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  visibility: ${(props) => (props.visible ? 'visible' : 'hidden')};
`;
export const CloseButton = styled.div`
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  padding: 15px;
  font-size: 18px;
  &::before {
    content: 'x';
  }
`;
const Modal = (props) => {
  return (
    <Wrapper visible={props.visible}>
      <CloseButton onClick={props.onClose} />
      {props.children}
    </Wrapper>
  );
};

export default Modal;
