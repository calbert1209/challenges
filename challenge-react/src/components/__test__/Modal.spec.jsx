import React from 'react';
import TestRenderer from 'react-test-renderer';
import Modal, { CloseButton } from '../Modal';
describe('Modal', () => {
  describe('render correctly', () => {
    it('visible true', () => {
      const testRenderer = TestRenderer.create(
        <Modal visible={true} onClose={jest.fn()} />
      );

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    it('visible false', () => {
      const testRenderer = TestRenderer.create(
        <Modal visible={false} onClose={jest.fn()} />
      );

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
  });
  it('click close', () => {
    const mockOnCloseProps = jest.fn();
    const testRenderer = TestRenderer.create(
      <Modal visible={true} onClose={mockOnCloseProps} />
    );
    const testInstance = testRenderer.root;
    testInstance.findByType(CloseButton).props.onClick();

    expect(mockOnCloseProps).toHaveBeenCalled();
  });
});
