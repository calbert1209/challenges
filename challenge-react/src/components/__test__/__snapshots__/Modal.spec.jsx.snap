// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render correctly visible false 1`] = `
<div
  className="sc-bdfBwQ kiuMva"
>
  <div
    className="sc-gsTCUz iXsQNC"
    onClick={[MockFunction]}
  />
</div>
`;

exports[`Modal render correctly visible true 1`] = `
<div
  className="sc-bdfBwQ gHXmzS"
>
  <div
    className="sc-gsTCUz iXsQNC"
    onClick={[MockFunction]}
  />
</div>
`;
